x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,574 @@
+a <<restart-cluster-rolling,rolling restart>> on your cluster is safer. An in-place update avoids some
+complications of a rolling restart, but incurs the following risks:
+
+* If you use PEM files, your certificate and key are in separate files. You
+must update both files _simultaneously_ or the node might experience a temporary
+period where it cannot establish new connections.
+* Updating the certificate and key does not automatically force existing
+connections to refresh. This means that even if you make a mistake, a node can
+seem like it's functioning but only because it still has existing connections.
+It's possible that a node will be unable to connect with other nodes, rendering
+it unable to recover from a network outage or node restart.
+****
+
+[[update-node-certs-same]]
+=== Update security certificates with the same CA
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+If your CA hasn't changed, you can replace your existing certificate on each
+node with a new certificate that has the same filename. When {es} restarts, it
+reloads the file and uses the new certificate. Using the same filename means
+that you don't have to update the `elasticsearch.yml` file, and can just
+replace the existing certificate.
+
+You don't have to restart each node, but doing so forces new TLS connections and is <<use-rolling-restarts,a recommended practice>> when updating certificates.
+Therefore, the following steps include a node restart after updating each
+certificate.
+
+You'll generate new node certificates and keys for both the transport layer and
+the HTTP layer.
+
+[[cert-password-updates]] 
+IMPORTANT: If your keystore is password protected, the password
+is stored in the {es} keystore, _and_ the password needs to change, then you
+must perform a <<restart-cluster-rolling,rolling restart>> on your cluster. You
+must also use a different file name for the keystore so that {es} reloads the
+new file.
+
+This procedure assumes that the you have access to the CA certificate and key
+that was originally generated (or otherwise held by your organization) and used
+to sign the certificate that is currently in use. It also assumes that the
+clients connecting to {es} on the HTTP layer are configured to trust the CA
+certificate.
+
+TIP: If your CA has changed, complete the steps in 
+<<update-node-certs-different,update security certificates with a different CA>>.
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
+Create a new PKCS#12 file that contains a certificate and key and replace
+the existing PKCS#12 file (such as `elastic-certificates.p12`) with the new
+certificate. Stop one node at a time, replace the node certificate, and then
+restart the node.
+
+NOTE: The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+. Open the `ES_PATH_CONF/elasticsearch.yml` file and check the names of the
```

Comment:
I was wondering. If this is a production cluster, then users should probably do all the steps that are dictated in our rolling restart guide _plus_ the change in the keystore we mention here ( rather than simply restarting the node ) in order to minimize service disruption. Should we try to find a way to tie into the rolling restart process by referencing steps or pointing to that doc more prominently?

Reply:
As Tim outlined in this [discuss forum post](https://discuss.elastic.co/t/updating-node-certificate-on-existing-cluster/243413/2):

> - Follow all the steps of a rolling restart.
> - While each node is stopped (the _Perform any needed changes step_), switch the node's SSL certificate and key. 

So I think that this approach can work:

1. Instruct users to complete all of the necessary steps to generate the keystores and certificates that they'll need.
2. Where we currently tell users to stop Elasticsearch, instruct them to begin a [rolling restart](https://elasticsearch_73781.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/restart-cluster.html#restart-cluster-rolling), but stop at step 5 for _Perform any needed changes_.
3. Replace the keystore files and certificates.
4. Complete the remaining steps for performing a rolling restart.

Does that seem like the right approach?

Reply:
Yes, I think that should be the safest option for our users!

Reply:
I incorporated the rolling restart steps in c82ffdb82049d4dd5a25c8a12bb9839220e62e58